New Entry and Terms: Python modules (#926)
Codecademy/docs: content/python/concepts/modules/terms/tqdm/tqdm.md

---
Title: 'tqdm'
Description: 'A module that allows the generation of progress bars in Python.'
Subjects:
  - 'Computer Science'
  - 'Data Science'
Tags:
  - 'Command Line'
  - 'Libraries'
  - 'Modules'
  - 'Scope'
CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
---

> The **`tqdm`** module allows for the generation of progress bars in Python. The name is derived from the Arabic word "taqaddum", which translates as "progress." It is designed to have minimal overhead, using algorithms to predict remaining time and to skip unnecessary iteration displays. It has about a 60ns (nanoseconds) overhead per iteration whereas other progress bar implementations can run a 800ns overhead per iteration. `tqdm` does not require any dependencies.

Reviewer: ```suggestion
The **`tqdm`** module allows for the generation of progress bars in Python. The name is derived from the Arabic word "taqaddum", which translates as "progress." It is designed to have minimal overhead, using algorithms to predict the remaining time and to skip unnecessary iteration displays. It has about a 60ns (nanoseconds) overhead per iteration whereas other progress bar implementations can run an 800ns overhead per iteration. `tqdm` does not require any dependencies.
```

> The **`tqdm`** module allows for the generation of progress bars in Python. The name is derived from the Arabic word "taqaddum", which translates as "progress." It is designed to have minimal overhead, using algorithms to predict the remaining time and to skip unnecessary iteration displays. It has about a 60ns (nanoseconds) overhead per iteration whereas other progress bar implementations can run an 800ns overhead per iteration. `tqdm` does not require any dependencies.

Reviewer: ```suggestion
The **`tqdm`** module allows for the generation of progress bars in Python. The name is derived from the Arabic word, "taqaddum," which translates as "progress." It is designed to have minimal overhead, using algorithms to predict the remaining time and to skip unnecessary iteration displays. It has about a 60ns (nanoseconds) overhead per iteration whereas other progress bar implementations can run an 800ns overhead per iteration. `tqdm` does not require any dependencies.
```

> > **Note:** `tqdm` does require running in an environment that supports carrage return (`\r`) and line feed (`\n`) control characters.

Reviewer: ```suggestion
> **Note:** `tqdm` does require running in an environment that supports carriage return (`\r`) and line feed (`\n`) control characters.
```

> > **Note:** `tqdm` does require running in an environment that supports carriage return (`\r`) and line feed (`\n`) control characters.

Reviewer: ```suggestion
> **Note:** `tqdm` requires an environment that supports carriage return (`\r`) and line feed (`\n`) control characters.
```
